### ISSUE TITLE
problem with continue training 

### ISSUE THREAD

**Issue by wingsyuan** [marian-nmt/marian-dev#550]
Hi,
@emjotde @frankseide @snukky @ugermann
when I start contiue training with the following setting, error occur:
[2019-11-15 11:32:31] Loading model from 150w_un_pretrain_60k/ens/model.npz.orig.npz
[2019-11-15 11:32:32] Loading model from 150w_un_pretrain_60k/ens/model.npz.orig.npz
[2019-11-15 11:32:34] Loading Adam parameters from 150w_un_pretrain_60k/ens/model.npz.optimizer.npz
[2019-11-15 11:32:35] [memory] Reserving 150 MB, device gpu0
[2019-11-15 11:32:36] [memory] Reserving 150 MB, device gpu1
[2019-11-15 11:32:36] [memory] Reserving 150 MB, device gpu2
[2019-11-15 11:32:36] [memory] Reserving 150 MB, device gpu3
[2019-11-15 11:32:37] [training] Model reloaded from 150w_un_pretrain_60k/ens/model.npz
[2019-11-15 11:32:37] [data] Restoring the corpus state to epoch 25, batch 96000
[2019-11-15 11:32:37] [data] Shuffling data
[2019-11-15 11:32:37] [data] Done reading 312394 sentences
[2019-11-15 11:32:38] [data] Done shuffling 312394 sentences to temp files
[2019-11-15 11:33:14] Error: attempted to wait for futureBufferedBatches_ when none pending
[2019-11-15 11:33:14] Error: Aborted from marian::data::BatchGenerator<DataSet>::BatchPtr marian::data::BatchGenerator<DataSet>::next() [with DataSet = marian::data::CorpusBase; marian::data::BatchGenerator<DataSet>::BatchPtr = std::shared_ptr<marian::data::CorpusBatch>] in /data/home/zhangshaoyuan/mt/marian-dev/src/data/batch_generator.h:242

[CALL STACK]
[0xd07b59]          marian::data::BatchGenerator<marian::data::CorpusBase>::  next  () + 0x44d
[0xcdc066]          marian::data::BatchGenerator<marian::data::CorpusBase>::  restore  (std::shared_ptr<marian::TrainingState>) + 0x22e
[0xcaffd3]          marian::Train<marian::SyncGraphGroup>::  run  ()   + 0x8dd
[0xc38d95]          mainTrainer  (int,  char**)                        + 0x452
[0xc3ac15]          main                                               + 0x1c9
[0x7f30307a7830]    __libc_start_main                                  + 0xf0
[0xc369f9]          _start                                             + 0x29

fur.sh: line 28: 10722 Aborted                 (core dumped) 
  
    
    $MARIAN_TRAIN \
        --model model_pretrain_finetune/ens/model.npz --type transformer \
        --train-sets /data/home/s/mt/dataset/muti-domain/dataset/muti_domain/news/parallel/train/news_train_tok_bpe.en /data/home/s/mt/dataset/muti-domain/dataset/muti_domain/news/parallel/train/news_train_tok_bpe.zh \
        --max-length 200 \
        --vocabs model_pretrain_finetune/vocab.mix.enzh.yml model_pretrain_finetune/vocab.mix.enzh.yml \
        --mini-batch-fit -w $WORKSPACE --mini-batch 1000 --maxi-batch 1000 \
        --valid-freq 500 --save-freq 500  --disp-freq 100 \
        --valid-metrics ce-mean-words perplexity translation \
        --valid-sets /data/home/s/mt/dataset/muti-domain/dataset/muti_domain/news/parallel/dev/dev1/newstest2018_tok_bpe_80.en /data/home/s/mt/dataset/muti-domain/dataset/muti_domain/news/parallel/dev/dev1/newstest2018_tok_bpe_80.zh \
        --valid-script-path "bash ../scripts/validate.sh" \
        --valid-translation-output model_pretrain_finetune/vali.output --quiet-translation \
        --beam-size 12 --normalize=1 \
        --valid-mini-batch 64 \
        --overwrite --keep-best \
        --early-stopping 10 --after-epochs $EPOCHS --cost-type=ce-mean-words \
        --log model_pretrain_finetune/ens$i/train.log --valid-log model_pretrain_finetune/ens$i/valid.log \
        --enc-depth 6 --dec-depth 6 \
        --tied-embeddings-all \
        --transformer-dropout 0.1 --label-smoothing 0.1 \
        --learn-rate 0.0003 --lr-warmup 16000 --lr-decay-inv-sqrt 16000 --lr-report \
        --optimizer-params 0.9 0.98 1e-09 --clip-norm 5 \
        --devices $GPUS --sync-sgd --seed $i$i$i$i  \
        --exponential-smoothing 

when I  recompiled with Release and tried again and it failed too.. 
I haved tried sevel way, but still failed..why??? what happened?
Can you give some advice? thanks  very much!!!

**Comment by snukky** [comment]
Do you change the training data when you restart the training? If so, try adding `--no-restore-corpus`. Maybe you can find more information in these threads: 
- https://groups.google.com/forum/#!topic/marian-nmt/ZvzvMN48xtI 
- https://github.com/marian-nmt/marian/issues/224

**Comment by wingsyuan** [comment]
@snukky 
fixed! thanks a lot!
